File: src/main/java/ru/avem/posum/utils/StatusBarLine.java
package ru.avem.posum.utils;

import javafx.application.Platform;
import javafx.scene.control.Label;
import javafx.scene.control.ProgressIndicator;
import javafx.scene.paint.Color;
import org.controlsfx.control.StatusBar;

import static java.lang.Thread.sleep;

public class StatusBarLine {
    private Label checkIcon;
    private boolean isMainView;
    private boolean isProcessView;
    private ProgressIndicator progressIndicator;
    private StatusBar statusBar;
    private Thread statusBarThread;
    private Label warningIcon;

    public StatusBarLine(Label checkIcon, boolean isMainView, ProgressIndicator progressIndicator,
                         StatusBar statusBar, Label warningIcon) {
        this.checkIcon = checkIcon;
        this.isMainView = isMainView;
        this.progressIndicator = progressIndicator;
        this.statusBar = statusBar;
        this.warningIcon = warningIcon;
    }

    public void setStatusOfProgress(String text) {
        clearStatusBar();
        Platform.runLater(() -> {
            hideIcons();
            statusBar.setStyle("-fx-padding: 0 0 0 3.2;");
            statusBar.setText(text);
            handleStatusBar();
        });
    }

    private void hideIcons() {
        checkIcon.setStyle("-fx-opacity: 0;");
        warningIcon.setStyle("-fx-opacity: 0;");
    }

    private void handleStatusBar() {
        if (statusBarThread != null) {
            statusBarThread.interrupt();
        }
        startNewStatusBarThread();
    }

    private void startNewStatusBarThread() {
        statusBarThread = new Thread(() -> {
            try {
                sleep(5000);
                clearStatusBar();
            } catch (InterruptedException ignored) {
            }
        });

        statusBarThread.start();
    }

    public void setStatus(String text, boolean isStatusOk) {
        clearStatusBar();
        Platform.runLater(() -> {
            statusBar.setText(text);
            initIcons(isStatusOk);
            handleStatusBar();
        });
    }

    private void initIcons(boolean isStatusOk) {
        checkIcon.setTextFill(Color.web("#009700"));
        warningIcon.setTextFill(Color.web("#D30303"));
        String padding;

        if (isStatusOk) {
            checkIcon.setStyle("-fx-opacity: 1;");
            padding = isMainView ? "-fx-padding: 0 0 0 4;" : "-fx-padding: 0 0 0 1.1;";
        } else {
            warningIcon.setStyle("-fx-opacity: 1;");
            padding = isMainView ? "-fx-padding: 0 0 0 -1.1;" : "-fx-padding: 0 0 0 -1.9;";
        }

        if (!isProcessView) {
            statusBar.setStyle(padding);
        }
    }

    public void clearStatusBar() {
        Platform.runLater(() -> {
            hideIcons();
            statusBar.setText("");
        });
    }

    public void toggleProgressIndicator(boolean isHidden) {
        if (isHidden) {
            Platform.runLater(() -> progressIndicator.setStyle("-fx-opacity: 0;"));
        } else {
            Platform.runLater(() -> progressIndicator.setStyle("-fx-opacity: 1;"));
        }
    }

    public void setProcessView(boolean processView) {
        isProcessView = processView;
    }
}

File: src/main/java/ru/avem/posum/utils/StatusBarLine.kt
package ru.avem.posum.utils

import javafx.application.Platform
import javafx.scene.control.Label
import javafx.scene.control.ProgressIndicator
import javafx.scene.paint.Color
import org.controlsfx.control.StatusBar

class StatusBarLine(private val checkIcon: Label, private val isMainView: Boolean,
                    private val progressIndicator: ProgressIndicator, private val statusBar: StatusBar,
                    private val warningIcon: Label) {
    private var statusBarThread = Thread()

    fun setStatusOfProgress(text: String) {
        clear()
        toggleProgressIndicator(false)
        Platform.runLater {
            statusBar.style = "-fx-padding: 0 0 0 3.2;"
            statusBar.text = text
            handleStatusBar()
        }
    }

    fun clear() {
        Platform.runLater {
            statusBar.text = ""
            hideIcons()
        }
    }

    private fun hideIcons() {
        checkIcon.style = "-fx-opacity: 0;"
        warningIcon.style = "-fx-opacity: 0;"
    }

    fun toggleProgressIndicator(isHidden: Boolean) {
        Platform.runLater {
            if (isHidden) progressIndicator.style = "-fx-opacity: 0;"
            else progressIndicator.style = "-fx-opacity: 1;"
        }
    }

    private fun handleStatusBar() {
        statusBarThread.interrupt()
        startNewStatusBarThread()
    }

    private fun startNewStatusBarThread() {
        statusBarThread = Thread {
            Utils.sleep(5000)
            clear()
        }
        statusBarThread.start()
    }

    fun setStatus(text: String, isStatusSuccessful: Boolean) {
        clear()
        toggleProgressIndicator(true)
        Platform.runLater {
            statusBar.text = text
            initIcons(isStatusSuccessful)
            handleStatusBar()
        }
    }

    private fun initIcons(isStatusSuccessful: Boolean) {
        checkIcon.textFill = Color.web("#009700")
        warningIcon.textFill = Color.web("#D30303")
        val padding: String

        if (isStatusSuccessful) {
            checkIcon.style = "-fx-opacity: 1;"
            padding = if (isMainView) "-fx-padding: 0 0 0 4;" else "-fx-padding: 0 0 0 1.1;"
        } else {
            warningIcon.style = "-fx-opacity: 1;"
            padding = if (isMainView) "-fx-padding: 0 0 0 -1.1;" else "-fx-padding: 0 0 0 -1.9;"
        }

        statusBar.style = padding
    }
}
